### ISSUE TITLE
Posting Snapshot Proposal Bug

### ISSUE THREAD

**Issue by alexyoung23j** [hicommonwealth/commonwealth#3236]

**Comment by alexyoung23j** [comment]
Having issues reproducing, waiting for more details from bug reporter.

**Comment by CowMuon** [comment]
If we don't get more details inside a full week, we should probably bounce back.

**Comment by alexyoung23j** [comment]
Looks like the issue was with a walletconnect signature- most likely not specifically related to snapshot integration but to the known walletconnect problems.

**Comment by alexyoung23j** [comment]
Received a report that snapshot proposal could not be created via metamask- appears original assumption that this was isolated to walletconnect was incorrect.

**Comment by CowMuon** [comment]
@alexyoung23j Yeah, TBH that was kind of an odd assumption, I didn't see the connection there. 

What I do see are at least 3 different bug reports: 
- #3264  -- [Timeless user unable to post](https://www.notion.so/buildcommon/3264-Not-able-to-post-on-snapshot-despite-having-the-required-token-amount-1c56453bdb794fde8bfbe37cfb3027d2)
- #3267 --  [Terra user unable to post thread ](https://www.notion.so/buildcommon/3267-Terra-user-unable-to-post-thread-38b422bb6f42400ab01bc6af11bd8484)
- [1Inch Unable to create Snapshot Proposal](https://www.notion.so/buildcommon/Unable-to-create-Snapshot-Proposal-on-1inch-f87e250bcf834bec9a498018974e477c)

Also- #3267 says Terra user unable to create proposal, but #3245 says Terra user unable to post thread. Which is it?

We need to get all of this sorted. We can use [Frack](https://commonwealth-frack.herokuapp.com/) for this. WalletConnect is there now, but let's first test on [React](https://commonwealth-react.herokuapp.com/) and then push any fixes to test (fixes should be on react branch only!) up to Frack.

**Comment by alexyoung23j** [comment]
In attempting to reproduce this problem, we may have uncovered a bug #3355 . Waiting for snapshot to respond to us on TG to confirm this fact.

**Comment by ianrowan** [comment]
I think I may have gotten to the bottom of this. Looking at snapshot's server source code https://github.com/snapshot-labs/snapshot-sequencer/blob/master/src/writer/proposal.ts

Their code 
![Image](https://user-images.githubusercontent.com/36428666/230218767-287ecf1e-4228-4e1e-a1d8-fd83e312a472.png)

One validation is the voting period, for 1inch this is 432000(5 days)  and for timelessfi this is 259200(3 days). We currently calculate a 4-day start - end date combo for all communities which would fail to validate for all proposals on both of these communities in.

I reproduced locally by setting my spaces period to 10 days and trying to create a proposal(worked before) and received a `Not Authorized` error as seen by the users with a 500 error from snapshot relating to the period. Hard coding a 10 day start - end period worked.

Solution: get voting delay(also not accounted for) and period from our space query and add to our object. When calculating start and end dates set:

`start = Date.now + space.voting.delay`
`end = start + space.voting.period`
That should account for all voting time configs 

@alexyoung23j @jnaviask @CowMuon
